mesa/datacollection.py

```diff
@@ -190,6 +241,34 @@ def get_reports(agent):
         )
         return agent_records
 
+    def _record_agenttype(self, model, agent_type):
+        """Record agent-type data in a mapping of functions and agents."""
+        rep_funcs = self.agenttype_reporters[agent_type].values()
+
+        def get_reports(agent):
+            _prefix = (agent.model.steps, agent.unique_id)
+            reports = tuple(rep(agent) for rep in rep_funcs)
+            return _prefix + reports
+
+        agent_types = model.agent_types
+        if agent_type in agent_types:
+            agents = model.agents_by_type[agent_type]
+        else:
+            from mesa import Agent
+
+            if issubclass(agent_type, Agent):
```

Comment:
why do this check only here when the model is already running. Would it not make more sense to check this during `__init__` (and so also only once)?

Reply:
Two reasons: This prevents a circular import, and most code paths won’t need to hit it at all.

Reply:
you could avoid circular imports in the exact same way in the init.